DALib/Drawing/Graphics.cs

```diff
@@ -193,43 +193,50 @@ public static SKImage RenderMap(MapFile map, DataArchive seoDat, DataArchive iaD
     ///     Renders a MapFile, given already extracted information
     /// </summary>
     /// <param name="map">
-    ///     The MapFile to render
+    ///     The <see cref="MapFile"/> to render
     /// </param>
     /// <param name="tiles">
-    ///     A collection of background tiles
+    ///     A <see cref="Tileset"/> representing a collection of background tiles
     /// </param>
     /// <param name="bgPaletteLookup">
-    ///     PaletteLookup for background tiles
+    ///     <see cref="PaletteLookup"/> for background tiles
     /// </param>
     /// <param name="fgPaletteLookup">
-    ///     PaletteLookup for foreground tiles
+    ///     <see cref="PaletteLookup"/> for foreground tiles
     /// </param>
     /// <param name="iaDat">
-    ///     IA archive for reading foreground tile files
+    ///     IA <see cref="DataArchive"/> for reading foreground tile files
+    /// </param>
+    /// <param name="foregroundPadding">
+    ///     The amount of padding to add to the height of the file and beginning rendering position
+    /// </param>
+    /// <param name="cache">
+    ///     A <see cref="MapImageCache"/> that can be reused to share <see cref="SKImageCache{TKey}"/> caches between multiple map renderings.
     /// </param>
     public static SKImage RenderMap(
         MapFile map,
         Tileset tiles,
         PaletteLookup bgPaletteLookup,
         PaletteLookup fgPaletteLookup,
-        DataArchive iaDat)
+        DataArchive iaDat, 
+        int foregroundPadding = 512,
+        MapImageCache? cache = null)
     {
-        const int FOREGROUND_PADDING = 512;
 
         //create lookups so we only render each tile piece once
         using var bgCache = new SKImageCache<int>();
         using var lfgCache = new SKImageCache<int>();
         using var rfgCache = new SKImageCache<int>();
 
-        //calculate width and height based on isometric view
-        var width = map.Width * CONSTANTS.TILE_WIDTH;
-        var height = map.Height * (CONSTANTS.TILE_HEIGHT + 1) + FOREGROUND_PADDING;
+        //calculate width and height
+        var width = (map.Width + map.Height + 1) * CONSTANTS.HALF_TILE_WIDTH;
```

Comment:
collateral correction for below correction

```suggestion
        var width = (map.Width + map.Height) * CONSTANTS.HALF_TILE_WIDTH;
```

Reply:
fixed